packages/cli-clean/src/clean.ts

```diff
@@ -28,13 +29,37 @@ type CleanGroups = {
 const DEFAULT_GROUPS = ['metro', 'watchman'];
 
 const rmAsync = promisify(rm);
+const rmAsyncOptions = {maxRetries: 3, recursive: true, force: true};
 
-function cleanDir(directory: string): Promise<void> {
-  if (!fileExists(directory)) {
-    return Promise.resolve();
-  }
+function isDirectoryPattern(directory: string): boolean {
+  return directory.endsWith('*') || directory.endsWith('?');
+}
+
+export async function cleanDir(directory: string): Promise<void> {
+  try {
+    if (isDirectoryPattern(directory)) {
+      const directories = await new Promise<string[]>((resolve, reject) => {
+        glob(directory, {}, (err, foundDirectories) => {
+          if (err) {
+            reject(err);
+          } else {
+            resolve(foundDirectories);
+          }
+        });
```

Comment:
doesn't glob have its promise interface?

Reply:
it does, but starting from 9.0, and we are currently using 7.x across the project 😢

Reply:
Let's upgrade it in a separate PR :D